### PR TITLE
Kong service config

### PR DESCRIPTION
**Pull request by mariya**
This PR:

- Configures Arteria services and authentication in Kong
- fixes an Nginx misconfiguration
- updates tarzan README

### CODE REVIEW

File: roles/tarzan/templates/kong.conf.j2 [hunk 1]
```diff
@@ -5,7 +5,7 @@ anonymous_reports = off
 #------------------------------------------------------------------------------
 # NGINX
 #------------------------------------------------------------------------------
-proxy_listen = 0.0.0.0:{{ tarzan_singularity_container_http_port }} http2, 0.0.0.0:{{ tarzan_singularity_container_https_port }} ssl
+proxy_listen = 0.0.0.0:{{ tarzan_singularity_container_http_port }}, 0.0.0.0:{{ tarzan_singularity_container_https_port }} ssl
```
Comment (b97pla) [inline roles/tarzan/templates/kong.conf.j2:8]: can and should we in fact disable the http port?

File: roles/tarzan/templates/kong.yml.j2 [hunk 2]
```diff
@@ -1,78 +1,41 @@
...
-#   config:
-#     hour: 2
-#   # tags are for your organization only and have no meaning for Kong:
-#   tags:
-#   - extra_limits
-#   - my_tag
+services:
+  - name: arteria_checksum
+    host: localhost
+    port: 10420
```
Comment (b97pla) [inline roles/tarzan/templates/kong.yml.j2:7]:
```suggestion
    port: "{{ arteria_checksum_port }}"
```

File: roles/tarzan/templates/kong.yml.j2 [hunk 3]
```diff
@@ -1,78 +1,41 @@
...
+    host: localhost
+    port: 10440
+    routes:
+    - paths:
+      - /arteria_delivery
+      strip_path: true
+
+  - name: arteria_archive
+    host: localhost
+    port: 10480
```
Comment (b97pla) [inline roles/tarzan/templates/kong.yml.j2:23]:
```suggestion
    port: "{{ archive_upload_port }}"
```

File: roles/tarzan/templates/kong.yml.j2 [hunk 4]
```diff
@@ -1,78 +1,41 @@
...
+    host: localhost
+    port: 10420
+    routes:
+    - paths:
+      - /arteria_checksum
+      strip_path: true
+
+  - name: arteria_delivery
+    host: localhost
+    port: 10440
```
Comment (b97pla) [inline roles/tarzan/templates/kong.yml.j2:15]:
```suggestion
    port: "{{ arteria_delivery_port }}"
```

File: roles/tarzan/templates/kong.yml.j2 [hunk 5]
```diff
@@ -1,78 +1,41 @@
...
+      strip_path: true
+
+plugins:
+  - name: key-auth
+    config:
+      key_names:
+      - apikey
+      key_in_body: false
+      key_in_header: false
+      key_in_query: true
```
Comment (b97pla) [inline roles/tarzan/templates/kong.yml.j2:36]: just double-checking whether we use `key_in_query` as it is now?

Comment (b97pla) [reply roles/tarzan/templates/kong.yml.j2:36]: perhaps we could add `hide_credentials: true` as well (the services don't need them, right?)

File: roles/tarzan/README.md [hunk 6]
```diff
@@ -1,17 +1,38 @@
...
+Finally, run the following command on the image file: `sha1sum /path-to-kong.sif`
 
-The resulting image should be uploaded to the path defined in `tarzan_singularity_image_file` in `defaults/main.yml`.
+Set `tarzan_singularity_image_sha1` to this value.
+
+# Certificates
+
+It is recommended to have a separate certificates for staging and production.
+
+On the node running Kong, run: `openssl req -x509 -newkey rsa:2048 -keyout tarzan_key.pem -out tarzan_cert.pem -days 1460 -nodes -subj '/CN=miarka2.uppmax.uu.se'`.
```
[the remark this answers is not in the document]
Comment (mariya) [reply roles/tarzan/README.md]: Oops, I actually forgot to commit a change that reflects this. Fixed now!

File: roles/tarzan/templates/kong.yml.j2 (hunk 5 above)
Comment (mariya) [reply roles/tarzan/templates/kong.yml.j2:36]: Yeah, we use `key_in_query` now. Fixed hide_credentials.

**Review by b97pla — APPROVED** [review]

**Pull request merged**